core/integration/steps/funding_payment_events.go

```diff
@@ -0,0 +1,149 @@
+	"fmt"
+	"time"
+
+	"code.vegaprotocol.io/vega/core/events"
+	"code.vegaprotocol.io/vega/core/integration/stubs"
+	"code.vegaprotocol.io/vega/logging"
+
+	"github.com/cucumber/godog"
+)
+
+func TheFollowingFundingPeriodEventsShouldBeEmitted(broker *stubs.BrokerStub, table *godog.Table) error {
+	fundingPeriodEvents := broker.GetFundingPeriodEvents()
+	for _, row := range parseFundingPeriodEventTable(table) {
+		fpe := FundingPeriodEventWrapper{
+			row: row,
+		}
+
+		matched := false
+		for _, evt := range fundingPeriodEvents {
+			if checkFundingPeriodEvent(evt, fpe) {
+				matched = true
+				break
+			}
+		}
+		if !matched {
+			start, _ := fpe.Start()
+			end, _ := fpe.End()
+			return fmt.Errorf("Funding period event with start=%v, end=%v, internal TWAP=%s, external TWAP=%s not found", start, end, fpe.InternalTWAP(), fpe.ExternalTWAP())
+		}
+	}
+	return nil
+}
+
+func DebugFundingPeriodEventss(broker *stubs.BrokerStub, log *logging.Logger) {
+	log.Info("DUMPING FUNDING PERIOD EVENTS")
+	data := broker.GetFundingPeriodEvents()
+	for _, evt := range data {
+		p := evt.Proto()
+		log.Infof("%s\n", p.String())
+	}
+}
+
+func VerifyTime(now time.Time, expected int64) error {
+	actual := now.Unix()
+	if actual == expected || now.UnixNano() == expected {
+		return nil
+	}
+	return fmt.Errorf("Expected unix time=%v, actual=%v", expected, actual)
+}
+
+func checkFundingPeriodEvent(evt events.FundingPeriod, row FundingPeriodEventWrapper) bool {
+	fundingPeriod := evt.FundingPeriod()
+
+	expectedStart, b := row.Start()
+	actualStart := fundingPeriod.GetStart()
+	if actualStart == 0 && b || !actualEqualsExpectedInSecondsOrNanos(actualStart, expectedStart) {
+		return false
+	}
+	expectedEnd, b := row.End()
+	actualEnd := fundingPeriod.GetEnd()
+	if actualEnd == 0 && b || !actualEqualsExpectedInSecondsOrNanos(actualEnd, expectedEnd) {
+		return false
+	}
+	expectedInternalTwap := row.InternalTWAP()
+	actualInternalTwap := fundingPeriod.GetInternalTwap()
+	if actualInternalTwap == "" && len(expectedInternalTwap) > 0 || expectedInternalTwap != actualInternalTwap {
+		return false
+	}
+	expectedExternalTwap := row.ExternalTWAP()
+	actualExternalTwap := fundingPeriod.GetExternalTwap()
+
+	if actualExternalTwap == "" && len(expectedExternalTwap) > 0 || expectedExternalTwap != actualExternalTwap {
+		return false
+	}
+	expectedFundingPayment, b := row.FundingPayment()
+	actualFundingPayment := fundingPeriod.GetFundingPayment()
+	if b && (actualFundingPayment == "" && len(expectedFundingPayment) > 0 || expectedFundingPayment != actualFundingPayment) {
+		return false
+	}
+	expectedFundingRate, b := row.FundingRate()
+	actualFundingRate := fundingPeriod.GetFundingRate()
+	if b && (actualFundingRate == "" && len(expectedFundingRate) > 0 || expectedFundingRate != actualFundingRate) {
+		return false
+	}
+
+	return true
+}
+
+func actualEqualsExpectedInSecondsOrNanos(actual, expected int64) bool {
+	return expected == actual || expected*1000000000 == actual
```

Comment:
Instead of magical constant number, just `expected * int64(time.Second)` makes it more clear